### PR TITLE
use faster sha256 and blake2b implementations

### LINKED ISSUE #61: Faster sha256 and blake2b implementations
https://github.com/minio/sha256-simd

>  Accelerate SHA256 computations in pure Go using AVX512 and AVX2 for Intel and ARM64 for ARM. On AVX512 it provides an up to 8x improvement (over 3 GB/s per core) in comparison to AVX2.
>
> This package is designed as a replacement for crypto/sha256. For Intel CPUs it has two flavors for AVX512 and AVX2 (AVX/SSE are also supported). For ARM CPUs with the Cryptography Extensions, advantage is taken of the SHA2 instructions resulting in a massive performance improvement.
>
> This package uses Golang assembly. The AVX512 version is based on the Intel's "multi-buffer crypto library for IPSec" whereas the other Intel implementations are described in "Fast SHA-256 Implementations on Intel Architecture Processors" by J. Guilford et al.

https://github.com/minio/blake2b-simd

> This package was initially based on the pure go BLAKE2b implementation of Dmitry Chestnykh and merged with the (cgo dependent) AVX optimized BLAKE2 implementation (which in turn is based on the official implementation. It does so by using Go's Assembler for amd64 architectures with a golang only fallback for other architectures.
> 
> In addition to AVX there is also support for AVX2 as well as SSE. Best performance is obtained with AVX2 which gives roughly a 4X performance increase approaching hashing speeds of 1GB/sec on a single core.

### PR DESCRIPTION
**Pull request by Stebalien**
fixes #61

### CODE REVIEW

File: sum.go [hunk 1]
```diff
@@ -115,6 +112,32 @@ func isBlake2b(code uint64) bool {
...
+var blake2b384Config = &blake2b.Config{Size: blake2b384Size}
+
+func sumBlake2b384(data []byte) []byte {
+	hasher, err := blake2b.New(blake2b384Config)
+	if err != nil {
+		panic(err)
+	}
+	hasher.Write(data)
+	var out [blake2b384Size]byte
+	copy(out[:], hasher.Sum(nil)[:blake2b384Size])
```
Comment (lgierth) [inline sum.go]: Why the reslice of the destination here?

**Comment by lgierth** [comment]
> blake2b-simd and sha256-simd are Apache-2.0-licensed, which can be used with our mostly MIT-licensed codebases fine.

Could you fill out the `license` field in the gxed packages though?

File: sum.go (hunk 1 above)
Comment (Stebalien) [reply sum.go]: I was worried that the hasher would over allocate. However, this turns out to not be the case so I'll correct it.

**Comment by Stebalien** [comment]
> Could you fill out the license field in the gxed packages though?

Done.

**Pull request merged**